pkg/providers/provider.go

```diff
@@ -48,4 +48,6 @@ type TemplateBuilder interface {
 	CPMachineTemplateName(clusterName string) string
 }
 
-type MachineConfig interface{}
+type MachineConfig interface {
```

Comment:
This interface was mainly created along with the `MachineConfigs` method to make it easier to output the yaml without having to access any of the fields. It seems like for this case, we just need to retrieve the OSFamily, so this looks good to me if we want to generalize how we are retrieving the values from MachineConfig.